developing/vcs/README.md

```diff
@@ -2,7 +2,321 @@
+right before merging, but that forces you to delete local branches with `-D`,
+which we don't think anyone should have to do.
+
+#### Rebase
+
+Pros:
+
+* Cleaner, more meaningful, easier to understand commit history
+* Easier to keep only commits that pass tests, which makes debugging
+easier with tools like `git bisect`.
+* Safer and easier to handle conflicts when commits are squashed
+* Easier to revert bad commits
+* Easier to change commit messages in a series of commits. We value great commit
+messages, so it should be easy to improve any message in a series of commits.
+* Allows reordering commits for more meaningful grouping. For example, you might
+have a PR with several commits that each update a dependency. If the tests fail
+due to a dependency in the middle, you'll want to fix that in a separate commit,
+then rebase interactively to move the fix under the dependency commit and
+combine it with `fixup`, then edit the commit message to explain what had to
+change after updating the dependency.
+
+Cons:
+
+* Riskier since it requires a force push
+* Rewrites history
+
+Ways to mitigate cons:
+
+* Protect the master branch from force pushes
+* Agree on a process where once a PR is under active review, no force pushes
+until the PR is approved and ready to be merged.
+* Original history can be viewed with `git reflog`
+
+#### Examples of popular repos that enforce rebase and squash
+
+* [Rails](https://edgeguides.rubyonrails.org/contributing_to_ruby_on_rails.html#squashing-commits)
+* [Django](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/working-with-git/#rebasing-branches)
+* [Golang](https://golang.org/doc/contribute.html#make_branch)
+* [Homebrew](https://docs.brew.sh/How-To-Open-a-Homebrew-Pull-Request#following-up)
+* [Hugo](https://github.com/gohugoio/hugo/blob/master/CONTRIBUTING.md#code-contribution-guidelines)
+* [React](https://github.com/facebook/react)
+* [Bootstrap](https://github.com/twbs/bootstrap/blob/master/.github/CONTRIBUTING.md#pull-requests)
+* [Electron](https://electronjs.org/docs/development/pull-requests#step-6-rebase)
+* [Node](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#step-5-rebase)
+
+#### Examples of popular repos that enforce merge
+
+I'm trying to avoid confirmation bias, but I haven't found any yet. If you know
+of any, please add them. I found some that don't enforce any specific workflow,
+and most of them say they will squash your commits via the GitHub interface.
+
+### Typical workflow
+
+Clone the repo locally (example below uses SSH):
+
+```
+git clone git@github.com:trussworks/Engineering-Playbook.git
+cd Engineering-Playbook
+```
+
+Create a local feature branch based off master. Start the branch name with your
+initials. For example: `mb-git-worfklow`. If the branch is related to a Pivotal
+story, add the story's ID to the end of the branch: `mb-git-workflow-166068559`
```

Comment:
Why the ceremony around initials and story ID?

Reply:
Having the story id in the branch allows some convenient integration with the tracker; your branch (and any PRs sent from it) will show up in the story automagically.

Initials I think are meant to mark branch ownership... imho git itself is opinionated in that branches are not owned by anybody. But I do this usually since folks seem to like it and not much (anything?) is lost

Reply:
It also helps to quickly identify who to ping if there are stale branches in GitHub.

Reply:
GitHub shows branch authors in the UI, so initials aren't needed anymore for cleanup purposes. Since branches can be collaborative, I prefer having a good name without any ownership marker.

<img width="383" alt="image" src="https://user-images.githubusercontent.com/162998/67134702-5eef3900-f1c8-11e9-9069-d728d4a12a58.png">